entities/storobj/storage_object.go

```diff
@@ -234,19 +234,28 @@ func FromBinaryOptional(data []byte,
 		}
 	}
 
-	if rw.Position < uint64(len(rw.Buffer)) {
-		multiVectorsLength := rw.ReadUint32()
-		if len(addProp.MultiVectors) > 0 {
-			if multiVectorsLength > 0 {
-				multivectors, err := rw.CopyBytesFromBuffer(uint64(multiVectorsLength), nil)
-				if err != nil {
-					return nil, errors.Wrap(err, "Could not copy multivectors")
-				}
-				if err := msgpack.Unmarshal(multivectors, &ko.MultiVectors); err != nil {
-					return nil, errors.Wrap(err, "Could not unmarshal multivectors")
-				}
-			}
+	if rw.Position < uint64(len(rw.Buffer)) && len(addProp.Vectors) > 0 {
+		vectorNamesToUnmarshal := map[string]interface{}{}
+		for _, name := range addProp.Vectors {
+			vectorNamesToUnmarshal[name] = nil
+		}
+		multiVectors, err := unmarshalMultiVectors(&rw, vectorNamesToUnmarshal)
+		if err != nil {
+			return nil, err
 		}
+		ko.MultiVectors = multiVectors
+
+		// if multiVectors != nil {
+		// 	// If parseObject is called, ko.Object will be overwritten making this effectively a
+		// 	// no-op, but I'm leaving it here to match the target vector behavior.
+		// 	if ko.Object.Vectors == nil {
+		// 		ko.Object.Vectors = make(models.Vectors)
+		// 	}
+		// 	for vecName, vec := range multiVectors {
+		// 		// assume at this level target vectors and multi vectors won't have the same name
+		// 		ko.Object.Vectors[vecName] = vec
```

Comment:
here, we can't assign `vec` because `models.Vector` is only a `[]float32` (does not support `[][]float32`. Should we just not set this (I don't know if we need to set it or not) or should we update `models.Vector` to support `[][]float32` like we did on `1.29` (i think this is a ton of work)?